src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -106,18 +105,17 @@ seastar::future<> ClientRequest::start()
               return with_blocking_future_interruptible<IOInterruptCondition>(
                 pg.wait_for_active_blocker.wait());
             }).then_interruptible([this,
-                                   has_pg_op,
                                    pgref=std::move(pgref)]() mutable {
               if (m->finish_decode()) {
                 m->clear_payload();
               }
-              return (has_pg_op ?
+              return (is_pg_op() ?
```

Comment:
yeah, just realized that it is `MOSDOp::finish_decode()` which fills up `m->ops`. wanted to cache the result `is_pg_op()`.

but since we want to decide whether to guard `interruptible_do_op()` with `with_sequencer()`, and `MOSDOp::finish_decode()` is called right in the guarded continuation chain, shall we just extract 
```c++
if (m->finish_decode()) {
  m->clear_payload();
}
```
out of the last continuation in the chain and put it before 
```c++
auto interruptible_do_op = interruptor::wrap_function([=] {
```
so that 

1. `m->ops` is ready before`is_pg_op()` is called, and 
2. we can cache the result of the call

?